ports/stm32/mboot/mboot_pack_dfu.py

```diff
@@ -231,8 +239,21 @@ def verify_pack_dfu(keys, filename):
             if header[1] == MBOOT_PACK_CHUNK_FW_GZIP:
                 chunk = zlib.decompress(chunk, wbits=-15)
             full_sig.update(chunk)
+            assert addr == base_addr + len(binary_data)
+            binary_data += chunk
+
     full_sig_pass = full_sig.final_verify(actual_sig, keys.sign_pk)
     assert full_sig_pass
+    return [{"address": base_addr, "data": binary_data}]
+
+
+def unpack_dfu(keys, args):
+    # Load previously generated keys.
+    keys.load()
+
+    # build a DFU file from the decrypted binary data
+    data = verify_pack_dfu(keys, args.infile[0])
+    dfu.build(args.outfile[0], [[{"address": data[0]["address"], "data": data[0]["data"]}]])
```

Comment:
much simpler to do: `dfu.build(args.outfile[0], [data])`

Reply:
Thanks, done